### PR TITLE
Fix documentation for referenced  apiregistration.k8s.io | v1 API fields

### PR DESCRIPTION
**Pull request by indevi**
Signed-off-by: Shivam Tyagi <mergesuccessful@gmail.com>

#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Fixes ##113438

#### Does this PR introduce a user-facing change?

This PR changes API field references on API docs. It uses exact field name instead of Golang type names.

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
The source changes look OK to me but you need to run `hack/update-codegen.sh`.

**Comment by MikeSpreitzer** [comment]
@ShivamTyagi12345 : would you like to follow up here?

**Comment by ShivamTyagi12345** [comment]
> @ShivamTyagi12345 : would you like to follow up here?

If you can help me with the next steps neccessary then I might be able to get this done quickly

**Comment by MikeSpreitzer** [comment]
```release-note
Corrected capitalization of some field names in the comments on API types in  apiregistration.k8s.io to match how they appear in JSON.
```

File: staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go [hunk 1]
```diff
@@ -115,12 +115,12 @@ const (
...
 type APIServiceCondition struct {
-	// Type is the type of the condition.
+	// type is the type of the condition.
 	Type APIServiceConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=APIServiceConditionType"`
-	// Status is the status of the condition.
+	// status is the status of the condition.
 	// Can be True, False, Unknown.
 	Status ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
-	// Last time the condition transitioned from one status to another.
+	// last time the condition transitioned from one status to another.
```
Comment (liggitt) [inline staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go:123]:
```suggestion
	// lastTransitionTime is the last time the condition transitioned from one status to another.
```

File: staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go [hunk 2]
```diff
@@ -115,12 +115,12 @@ const (
...
 	// +optional
 	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,3,opt,name=lastTransitionTime"`
 	// Unique, one-word, CamelCase reason for the condition's last transition.
```
Comment (liggitt) [inline staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go:126]:
```suggestion
	// reason is a unique, one-word, CamelCase reason for the condition's last transition.
```

Comment (liggitt) [reply staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go:126]: also update `message` below

File: staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go [hunk 3]
```diff
@@ -155,8 +155,8 @@ type APIService struct {
 	// +optional
```
Comment (liggitt) [inline staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go:155]: update the `conditions` field doc above

**Pull request closed without merging**